system.go

```diff
@@ -30,6 +30,21 @@ import (
 	_ "github.com/elastic/go-sysinfo/providers/windows"
 )
 
+type ProviderOption func(*registry.ProviderOptions)
+
+// WithHostFS returns a provider with a custom HostFS root path,
+// enabling use of the library from within a container, or an alternate root path.
+// For example, WithHostFS("/hostfs") could be used when these mounts are present
+// in the container:
+//
+//   - /proc:/hostfs/proc
+//   - /sys/fs/cgroup:/hostfs/sys/fs/cgroup
```

Comment:
Considering  all things that are "HostFS" aware, are there any other mounts that a user would need to add? What about some combination of /etc/{os,lsb}-release for the OS detection?

The [machine ID code](https://github.com/elastic/go-sysinfo/blob/6057d34798d93384013c2e8f8e3bd0b00c599912/providers/linux/machineid.go#L30) isn't host FS aware, and think that creates an inconsistency for users. Do you think that should also be made aware of hostfs?

Reply:
Ah, good catch. Lemme poke around...

Reply:
Alright, added the hostfs arg to MachineID. Did some poking around, it seems like everything else in the linux provider should be be hostfs aware if it can't be. One exception is `NativeArchitecture()`, which shouldn't matter.

Reply:
Can you please list in the godoc all of the necessary mounts for this example to work.